lane.py

```diff
@@ -387,897 +372,812 @@ def has_next_page(self):
-                LicensePool.data_source==self.license_source
-            )
+    @hybrid_property
+    def target_age(self):
+        return self._target_age
 
-        if self.age_range != None:
-            if (Classifier.AUDIENCE_ADULT in self.audiences
-                or Classifier.AUDIENCE_ADULTS_ONLY in self.audiences):
-                # Books for adults don't have target ages. If we're including
-                # books for adults, allow the target age to be empty.
-                audience_has_no_target_age = work_model.target_age == None
-            else:
-                audience_has_no_target_age = False
-
-            if len(self.age_range) == 1:
-                # The target age must include this number.
-                r = NumericRange(self.age_range[0], self.age_range[0], '[]')
-                q = q.filter(
-                    or_(
-                        work_model.target_age.contains(r),
-                        audience_has_no_target_age
-                    )
-                )
-            else:
-                # The target age range must overlap this age range
-                r = NumericRange(self.age_range[0], self.age_range[-1], '[]')
-                q = q.filter(
-                    or_(
-                        work_model.target_age.overlaps(r),
-                        audience_has_no_target_age
-                    )
-                )
+    @target_age.setter
+    def set_target_age(self, value):
+        """Setting .target_age will lock .audiences to appropriate values.
 
-        if self.fiction == self.UNCLASSIFIED:
-            q = q.filter(work_model.fiction==None)
-        elif self.fiction != self.BOTH_FICTION_AND_NONFICTION:
-            q = q.filter(work_model.fiction==self.fiction)
+        If you set target_age to 16-18, you're saying that the audiences
+        are [Young Adult, Adult].
 
-        if self.media:
-            q = q.filter(edition_model.medium.in_(self.media))
+        If you set target_age 12-15, you're saying that the audiences are
+        [Adult, Children].
 
-        # TODO: Also filter on formats.
+        If you set target age 0-2, you're saying that the audiences are
+        [Children].
 
-        q = self.only_show_ready_deliverable_works(q, work_model)
+        In no case is the "Adults Only" audience allowed, since target
+        age only makes sense in lanes intended for minors.
+        """
+        if value is None:
+            self._target_age = None
+            return
+        audiences = []
+        if isinstance(value, int):
+            value = (value, value)
+        if isinstance(value, tuple):
+            value = tuple_to_numericrange(value)
+        self._target_age = value
+        if value.upper >= 18:
+            audiences.append(Classifier.AUDIENCE_ADULT)
+        if value.lower < Classifier.YOUNG_ADULT_AGE_CUTOFF:
+            audiences.append(Classifier.AUDIENCE_CHILDREN)
+        if value.upper >= Classifier.YOUNG_ADULT_AGE_CUTOFF:
```

Comment:
Should this also check that value.upper <= 18, or prevent setting target age > 18?